adhocracy4/follows/static/follows/FollowButton.jsx

```diff
@@ -18,73 +18,95 @@ const translated = {
   following: django.gettext('Following')
 }
 
-export const FollowButton = (props) => {
+export const FollowButton = ({ 
+  project, 
+  authenticatedAs, 
+  useMbStyles = false, 
+  alertTarget = null 
+}) => {
   const [following, setFollowing] = useState(null)
   const [alert, setAlert] = useState(null)
-
+  
   const followBtnText = following ? translated.following : translated.follow
-
   const followDescriptionText = following
     ? translated.followingDescription
     : translated.followDescription
 
-  const followAlertText = following
-    ? translated.followingAlert
-    : translated.followAlert
-
   useEffect(() => {
-    if (props.authenticatedAs) {
+    if (authenticatedAs) {
       api.follow
-        .get(props.project)
+        .get(project)
         .done((follow) => {
           setFollowing(follow.enabled)
-          setAlert(follow.alert)
         })
         .fail((response) => {
           if (response.status === 404) {
             setFollowing(false)
           }
         })
     }
-  }, [props.project, props.authenticatedAs])
+  }, [project, authenticatedAs])
 
   const removeAlert = () => {
     setAlert(null)
   }
 
   const toggleFollow = () => {
-    if (props.authenticatedAs === null) {
+    if (authenticatedAs === null) {
       window.location.href = config.getLoginUrl()
       return
     }
-    api.follow.change({ enabled: !following }, props.project).done((follow) => {
+    
+    api.follow.change({ enabled: !following }, project).done((follow) => {
       setFollowing(follow.enabled)
       setAlert({
-        type: 'success',
-        message: followAlertText
+        type: follow.enabled ? 'success' : 'warning',
+        message: follow.enabled ? translated.followAlert : translated.followingAlert
       })
     })
   }
 
+  const buttonClasses = following ? 'a4-btn a4-btn--following' : 'a4-btn a4-btn--follow'
+
+  const AlertPortal = () => {
+    if (!alert || !alertTarget) return null
+    
+    const container = document.getElementById(alertTarget)
+    if (!container) return null
```

Comment:
Maybe add a `console.error` or throw an error, I think that's better for devs to understand where the issue is then.